src/Stratis.SmartContracts.CLR/EmbeddedContractIdentifier.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Linq;
+using NBitcoin;
+
+namespace Stratis.SmartContracts.CLR
+{
+    public struct EmbeddedContractIdentifier
+    {
+        private const string signature = "Embedded";
+        private static byte[] embeddedContractSignature = System.Text.Encoding.ASCII.GetBytes(signature);
+        private uint160 value;
+
+        public EmbeddedContractIdentifier(uint160 id)
+        {
+            this.value = id;
+        }
+
+        public EmbeddedContractIdentifier(ulong contractTypeId, uint version)
+        {
+            this.value = new uint160(embeddedContractSignature.Concat(BitConverter.GetBytes(contractTypeId)).Concat(BitConverter.GetBytes(version)).ToArray());
```

Comment:
Instead of using a string to bytes what do you think of just setting the first 8 bytes to 0? So a system contract address in hex looks something like 0x0000000000000000....1....1. Main benefit would be easier for a human to distinguish just by looking at the identifier bytes in a block explorer.

Reply:
Good idea.